str/fine-mapping/mcv/files_prep_dosages.py

```diff
@@ -0,0 +1,182 @@
+    df.sort_index(axis=1, inplace=True)
+    df.reset_index(inplace=True)
+    return df
+
+
+def dosages(chrom,df_chr):
+    import pandas as pd
+    import numpy as np
+    from cpg_utils.hail_batch import output_path
+    from cyvcf2 import VCF
+    """
+    Extracts the dosage files for each gene in the df DataFrame.
+    """ ""
+
+    # Load metadata to extract the window around the gene
+    gene_info = pd.read_csv('gs://cpg-bioheart-test/tenk10k/saige-qtl/300libraries_n1925_adata_raw_var.csv')
+
+    # Load in the SNP
+
+    local_snp_file = 'local.snp.vcf.bgz'
+    local_snp_file_index = 'local.snp.vcf.bgz.tbi'
+
+    snp_file = to_path(
+        f'gs://cpg-bioheart-test/tenk10k/str/associatr/common_variant_snps/hail_filtered_{chrom}.vcf.bgz'
+    )
+    snp_file_index = to_path(
+        f'gs://cpg-bioheart-test/tenk10k/str/associatr/common_variant_snps/hail_filtered_{chrom}.vcf.bgz.tbi'
+    )
+
+    snp_file.copy(local_snp_file)
+    snp_file_index.copy(local_snp_file_index)
+    snp_vcf = VCF(local_snp_file)
+
+    # Load in the TR VCFs
+    local_tr_file = 'local.tr.vcf.bgz'
+    local_tr_file_index = 'local.tr.vcf.bgz.tbi'
+
+    tr_file = to_path(
+        f'gs://cpg-bioheart-test/tenk10k/str/associatr/final-freeze/input_files/tr_vcf/v1-chr-specific/hail_filtered_{chrom}.vcf.bgz'
+    )
+    tr_file_index = to_path(
+        f'gs://cpg-bioheart-test/tenk10k/str/associatr/final-freeze/input_files/tr_vcf/v1-chr-specific/hail_filtered_{chrom}.vcf.bgz.tbi'
+    )
+
+    tr_file.copy(local_tr_file)
+    tr_file_index.copy(local_tr_file_index)
+
+    tr_vcf = VCF(local_tr_file)
+
+    for gene in df_chr['gene_name'].unique():
+        if to_path(output_path(f'dosages/{chrom}/{gene}_dosages.csv')).exists():
+            print(f"Dosage file for {gene} already exists, skipping.")
+            continue
+
+        gene_ensg = gene
+        gene_info_filtered = gene_info[gene_info['gene_ids'] == gene_ensg]
+        chrom = gene_info_filtered.iloc[0]['chr']
+        start = int(gene_info_filtered.iloc[0]['start'])
+        end = int(gene_info_filtered.iloc[0]['end']) + 100_000
+        start_body = max(0, start - 100_000)
+
+        # Extract genotype matrices for tandem repeats and SNPs
+        tr_df = tr_extract_genotype_matrix(chrom, start_body, end,tr_vcf)
+        snp_df = snp_extract_genotype_matrix(chrom, start_body, end,snp_vcf)
+
+        # Merge the two genotype dfs together
+        variant_df = tr_df.merge(snp_df)
+        variant_df['sample'] = variant_df['sample'].astype(float)
+
+        # Save file
+        variant_df.to_csv(output_path(f"dosages/{chrom}/{gene_ensg}_dosages.csv"), header=True)
+
+
+@click.option('--estrs-path', type=str, required=True, help='Path to the estrs file.')
+@click.command()
+def main(estrs_path):
+    b = get_batch(name='Dosages files prep for SuSie MCV')
+    df = pd.read_csv(estrs_path)
+    df = df.drop_duplicates(subset=['gene_name', 'chr'])
+    # sort by chromosome
+    for chrom in ['chr8']:
```

Comment:
I know / assume you'll remove this chr filter for the full run, just flagging so I don't forget